### ISSUE TITLE
Line breaks not created/escaped correctly in WHERE clauses

### ISSUE THREAD

**Issue by larskn** [querydsl/querydsl#1275]
As described here: https://groups.google.com/forum/#!topic/querydsl/z08cNws-WSI

Using querydsl-sql 3.6.2 with MS SQL and MySQL for literal query generation. Simple example:
```java
private String createContainsQuery(final SQLTemplates template, final String filter) {
        final Configuration configuration = new Configuration(template);
        configuration.setUseLiterals(true);
        final SimplePath<Object> table = Expressions.path(Object.class, "Tablename");
        final StringPath column = Expressions.stringPath(table, "aColumn");
        final SQLQuery sqlQuery = new SQLQuery(configuration).from(table).where(column.contains(filter));
        return sqlQuery.getSQL(column).getSQL();
    }
```

Creating contains and equals statements with a line break ("foo\nbar") or a literal "\n" ("foo\\nbar") leads to the following reults:

### SQL Server 

* Contains with line break:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo\nbar%' escape '\'
```
This does not work since SQL server (at least 2014) does not seem to accept a literal "\n" as a line break, only **'%foo' + CHAR(13)  + CHAR(10) + 'bar%'** or **'%foo' + CHAR(10) + 'bar%'** (depending on the OS) seem to work.
 
* Contains with literal \n:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo\\nbar%' escape '\'
```
This does work although **'%foo\nbar%'** (without escape) would be sufficient
 
* Equals with line break:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn = 'foo\nbar'
```
* Equals with literal \n:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn = 'foo\nbar'
```
One query should be looking for a line break and the other should be looking for a literal "\n" but both produce the same output. (with the latter beeing correct)
 
### MySQL

* Contains with line break:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo\\nbar%'
```
The "\n" is escaped wrongly and thus the query is looking for the literal "\n" and not for a line break.
 
* Contains with literal \n:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo\\\\nbar%'
```
This query searches for the literal "\\\\n" (two backslashes) rather than the literal "\n"
 
* Equals with line break:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn = 'foo\\nbar'
```
* Equals with literal \n:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn = 'foo\\nbar'
 ```
Again both queries produce the same output although they should search for different results.

**Comment by larskn** [comment]
I looked a little more into the issue and spotted one thing I overlooked at first:

Looking at the first query I mentioned:
```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo\nbar%' escape '\'
```
Since the "\n" is showing up in the output means it was already escaped. Otherwise the query would look like.

```sql
select Tablename.aColumn
from Tablename
where Tablename.aColumn like '%foo
bar%' escape '\'
```

Which means the general problem seems to be that QueryDSL internally escapes the line break at some point which leads to the query looking for a literal line break.

If i only create the base query
```sql
select Tablename.aColumn
from Tablename
```
and manually add in Java code
```java
query += " where Tablename.aColumn like '%foo\nbar%' ";
```
The query correctly returns the row with the line break. (So CHAR(10) and CHAR(13) are not needed, this is handled by the JDBC driver)
